### PR TITLE
Kang2019decoupling 2+joshpapermaster

### PR DESCRIPTION
**Pull request by joshpapermaster**

### CODE REVIEW

File: _summaries/kang2019decoupling_2.md [hunk 1]
```diff
@@ -1,19 +1,86 @@
...
 
-TODO: Summarize the paper:
+<!-- TODO: Summarize the paper:
 * What is the core idea?
 * How is it realized (technically)?
 * How well does the paper perform?
-* What interesting variants are explored?
+* What interesting variants are explored? -->
+## What is the core idea?
+Visual recognition solutions can struggle when there is a long-tailed distribution, which is also known as the class imbalance problem. This paper separates the this problem into two processes: representation learning and classification.
```
Comment (timchen0618) [inline _summaries/kang2019decoupling_2.md:16]: could explain why they do this decoupling and elaborate more on what exactly is this decoupling.
(I cannot understand it reading your summary, and then I read the paper and I understand.)

File: _summaries/kang2019decoupling_2.md [hunk 2]
```diff
@@ -1,19 +1,86 @@
...
-* What interesting variants are explored?
+* What interesting variants are explored? -->
+## What is the core idea?
+Visual recognition solutions can struggle when there is a long-tailed distribution, which is also known as the class imbalance problem. This paper separates the this problem into two processes: representation learning and classification. 
+
+Representation learning
+- trained with different sampling techniques and losses
+
+Classification
+- the long tailed classes are recognized by the model from various classifiers
```
Comment (timchen0618) [inline _summaries/kang2019decoupling_2.md:22]: you could add that the representation learned is then used to train the classifiers

File: _summaries/kang2019decoupling_2.md [hunk 3]
```diff
@@ -1,19 +1,86 @@
...
+
+Representation learning
+- trained with different sampling techniques and losses
+
+Classification
+- the long tailed classes are recognized by the model from various classifiers  
+
+The paper determined the following findings from using their decoupled process:
+- instance based sampling provided really good generalized results
+- during representation learning it's beneficial to adjust decision boundaries of classifier
```
Comment (timchen0618) [inline _summaries/kang2019decoupling_2.md:26]: I don't think this is what they meant. They meant it's beneficial to adjust decision boundaries of classifier during the phase of classifier training. But the decision boundaries were originally learned during representation learning, which is the first stage of the process.

File: _summaries/kang2019decoupling_2.md [hunk 4]
```diff
@@ -1,19 +1,86 @@
...
+
+Representation Learning:
+- Instance-balanced sampling
+    - All examples have same chance of being sampled
+- Class-balanced sampling
+    - Each class has same chance of being sampled and then example uniformly sampled within the class
+- Square-root sampling
+    - Middle ground between the two previous sampling techniques. Square-roots the total number of images in each class when calculating the ratio of images in specific class / total images
+- Progressively-balanced sampling
+    - Uses instance-balanced sampling for some epochs and then switches to class-balanced sampling
```
Comment (timchen0618) [inline _summaries/kang2019decoupling_2.md:42]: I think they didn't do the hard-assignment version, they do the softer version where they interpolate these two kinds of sampling

File: _summaries/kang2019decoupling_2.md [hunk 5]
```diff
@@ -1,19 +1,86 @@
...
+- Loss re-weighting
+    - Not tested in this paper, but the approaches they did test produced better results
+
+Classification:
+- Classifier Re-training (cRT)
+    - Use class-balanced sampling and re-initialize and optimize the weights for a few epochs
```
Comment (timchen0618) [inline _summaries/kang2019decoupling_2.md:48]: first re-initialize the classifier weights, and then train for a few epochs using class-balanced sampling?

File: _summaries/kang2019decoupling_2.md [hunk 6]
```diff
@@ -1,19 +1,86 @@
...
+- Loss re-weighting
+    - Not tested in this paper, but the approaches they did test produced better results
+
+Classification:
+- Classifier Re-training (cRT)
+    - Use class-balanced sampling and re-initialize and optimize the weights for a few epochs 
+- Nearest Class Mean classifier (NCM)
+    - First get the mean feature representation for each class, then run nearest neighbors using cosine similarity or Euclidean distance
+- τ -normalized classifier (τ -normalized)
+    - Normalize the weights with a hyperparameter factor τ
```
Comment (timchen0618) [inline _summaries/kang2019decoupling_2.md:52]: can't really understand how to do it with your description, perhaps could just show the equation?

File: _summaries/kang2019decoupling_2.md [hunk 7]
```diff
@@ -1,19 +1,86 @@
...
+
+## How well does the paper perform?
+
+Experiments done on three datasets
+- Places-LT (artificially unbalanced from original)
+- ImageNet-LT (artificially unbalanced from original)
+- iNaturalist 2018 (natural)
+
+
+The many, medium, few, all titles below refer to the number of images trained on from the dataset
```
Comment (timchen0618) [inline _summaries/kang2019decoupling_2.md:64]: different classes? many means classes with more training images, few means classes with very few training images?

File: _summaries/kang2019decoupling_2.md [hunk 8]
```diff
@@ -1,19 +1,86 @@
...
+
+Results continue to show decoupled methods providing the best results
+![decoupling](kang2019decoupling_2d.png)
+![decoupling](kang2019decoupling_2e.png)
+
+## What interesting variants are explored?
+- Mentioned but didn't evaluate loss re-weighting in representation learning
+- No other variants besides all the different methods evaluated above 
 
 ## TL;DR
```
Comment (timchen0618) [inline _summaries/kang2019decoupling_2.md:83]: TLDR is well written!

**Comment by timchen0618** [comment]
Hi, can you see my reviews? If so, can you please take a look and consider changing some of your summary?

**Comment by joshpapermaster** [comment]
> Hi, can you see my reviews? If so, can you please take a look and consider changing some of your summary?

@timchen0618 Yep, thanks for the review. The revisions have been uploaded

**Review by zhaoyue-zephyrus — APPROVED** [review]
